syndeo-template/src/test/java/com/google/cloud/syndeo/it/PubsubDlqIT.java

```diff
@@ -146,35 +148,30 @@ public void testPubsubToPubsubSuccessful() throws Exception {
     // Publish correct messages to the source queue
     publishCorrectDataToPubsub();
 
-    // Wait for the Syndeo pipeline to move all the data.
-    Thread.sleep(3 * ONE_MINUTE_MILLIS);
-
-    // Wait for a while to check the reults in the pubsub sink
-    {
-      Instant start = Instant.now();
-      while (true) {
-        PullResponse response = pubsubResourceManager.pull(pubsubSinkSubscription, 100);
-        if (response.getReceivedMessagesCount() == MESSAGE_COUNT) {
-          // Test sucessful
-          break;
-        } else if (response.getReceivedMessagesCount() > MESSAGE_COUNT) {
-          throw new AssertionError(
-              String.format(
-                  "Expected at most %s elements to be inserted to Pubsub sink, but found %s.",
-                  MESSAGE_COUNT, response.getReceivedMessagesCount()));
-        } else if (java.time.Duration.between(start, Instant.now())
-                .compareTo(java.time.Duration.ofMinutes(20))
-            > 0) {
-          // If we spent over 20 minutes waiting for the job, then we must exit
-          throw new AssertionError(
-              String.format(
-                  "Expected %s elements to be inserted to Pubsub sink, but found %s after over 20 minutes.",
-                  MESSAGE_COUNT, response.getReceivedMessagesCount()));
-        } else {
-          // Iterate once more
-          Thread.sleep(ONE_MINUTE_MILLIS);
-        }
-      }
+    // Poll the pubsub sink until the messages are written
+    PubsubMessagesCheck pubsubCheck =
+        PubsubMessagesCheck.builder(pubsubResourceManager, pubsubSinkSubscription)
+            .setMinMessages(MESSAGE_COUNT)
+            .setMaxMessages(MESSAGE_COUNT)
+            .build();
+
+    await("Retrieving messages from Pub/Sub sink")
+        .atMost(Duration.ofMinutes(7))
+        .pollInterval(Duration.ofSeconds(15))
```

Comment:
nit: any reason to not make this smaller? e.g., 5s? it produces more logs, but 15s may be representative, specially if we decide to plug some of those tests on DirectRunner later (as we've done for other templates, it's not planned here). for reference, we have ITs being executed in the order of ~5-6s on the DirectRunner.